### PR TITLE
Fixes #31425 - Templates rendering host status

### PR DESCRIPTION
**Pull request by kamils-iRonin**
The idea is to save a status every time a template is rendered (except preview mode).  We would have information about which templates were rendered, with which hosts and if any errors occurred. Based on this data, we can determine the rendering status for a particular host.

### CODE REVIEW

File: app/subscribers/foreman/rendering/unsafemode_rendered_subscriber.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+                                              .update_all(unsafemode_status: HostStatus::RenderingStatusCombination::OK)
+
+        # refresh global statuses
+        hosts = Host::Managed.joins(:rendering_status_combinations)
+                             .where(rendering_status_combinations: { template_id: template_id })
+        RefreshGlobalStatuses.call(hosts)
+      end
+    end
+  end
+end
```
Comment (lzap) [inline app/subscribers/foreman/rendering/unsafemode_rendered_subscriber.rb:16]: Looks like all subscribers look quite similar and some code sharing might be appropriate.

**Comment by laugmanuel** [comment]
After some small tweaks (see force pushes), I could successfully test the functionality of the PR in my local test environment based on the current `develop` branch.

**Comment by lzap** [comment]
I like it, also introducing views is a good step forward now that we only support postgres. There are definitely some queries which could be optimized on the DB level.

Introduction of SQL views requires broader consensus, but I am all for it:

https://community.theforeman.org/t/introducing-sql-views-via-scenic-gem/22713

**Comment by ares** [comment]
Nice functionality and I think this last incarnation of the rendering status resolved all concerns. Once you want this to be reviewed, please drop the draft attribute.

**Comment by ares** [comment]
Hello @kamils-iRonin , given this is a draft for a very long time we'd like to know if you have a plan to move this out of the draft in a foreseeable future. This is a really great feature and it seems there was an activity in the last month. Could you please summarize what's the current status?

If we don't get a reply in 2 weeks, we'll close the PR but it can always be reopened.

**Comment by tbrisker** [comment]
2 weeks have passed, closing. Feel free to reopen when you get back to this.

**Pull request closed without merging**